### ISSUE TITLE
VR issue in 10.8.1 build 1907

### ISSUE THREAD

**Issue by Crrispy** [vpinball/vpinball#1291]
Hello,

VR starts fine in 10.8.1 now, but the resolution is bad, with lots of aliasing etc....
Just overwriting the files with 10.8.0 build 1836 and it's perfect again, with the same openvr dll (open composite with VDXR).
It does this on any table.

**Comment by thsa** [comment]
It seems that 'Preview Mode' resolution and and VR output are coupled now, no matter whether preview mode is 'Left Eye' or 'None'. I use no preview (black window) with very small size, e.g. width=216, height=384 and get a terrible VR resolution. Increasing preview to 2160x3840 (4k portrait monitor) results in a much better VR-resolution, but not enough for me, because I use a Pimax Crystal with about 3000x3000 pixels per eye. It looks like it uses preview mode width also for VR.

**Comment by Crrispy** [comment]
Interesting, I have preview set to 1920x1080, could explain my blurry/aliased VR image but not catastrophic.

**Comment by thsa** [comment]
Till today, this issue does not seem to be solved. Build 1978 (last 1.8.0) works perfectly for me. Since #1979 the VR view (within headset) is extremely blurred and not usable, if I have set the preview to 'none' or 'left eye' and preview size to something small as 400x600. Even if I put the preview on my vertical 4k monitor (left eye 2160x3840), the resolution within the headset is significantly lower than its physical resolution (2880x2880). Evidently, the preview resolution set for the monitor is a limiting factor for the headset. In #1978, the last working build, this was not the case.

**Comment by TurboHenk** [comment]
I can confirm this that this is still the case. I also have the Pimax Crystal and I run 4315x5100px in the most demanding tables and 1.25/1.33/1.5 supersampling in less demanding tables.

So 10.8.1. is still unusable for probably all VR users, because the resolution is simply completely off/incorrectly rendered.

It would be nice if this gets fixed because recent bugfixes now only seem to be applied to 1.0.8.1.

**Comment by thsa** [comment]
I retested with VPinballX_GL-10.8.1-2464-afcd5f8-Release-win-x64. I tested with both, the original openvr_api64.dll (SteamVR) and with the OpenComposite runtime dll (OpenXR). Both seem to work perfectly fine now: I get the expected full resolution and speed is absolutely smooth with a Pimax Crystal, 90 Hz on an RTX 3090.

**Comment by TurboHenk** [comment]
Thanks for your report and testing. I retested it and an old release too. YES huge difference(with 1.0 full resolution it works indeed), so what I see is still bugged and now I tested it a bit further and I see what WAS bugged in the previous 10.8.1. and what's bugged _now_ in the recent 10.8.1. 2464 release: 
- In the old 10.8.1. version it ran super low resolution in VR no matter what setting you used. 
- In the most recent 10.8.1. version supersampling simply doesn't work anymore. When I set supersampling to 1.5 there is a massive amount of shimmering on the table instead of an improvement. This while the image gets sharper with 10.8.0 while using supersampling. 

So this is probably a new bug but I'm not sure if it's reported already. I think that this issue should be closed but that a new issue should be made for "broken supersampling in VR"?

**Comment by Violins77** [comment]
Indeed, supersampling has been broken for me since 10.7 and never seemed to worked at more than 1.0 since then. There is already https://github.com/vpinball/vpinball/issues/199 but it was closed and never really resolved.

I will test tonight with the Windowed resolution size if it really makes a difference in the output.
